tee-worker/sidechain/rpc-handler/src/direct_top_pool_api.rs

```diff
@@ -114,18 +120,50 @@ where
 		Ok(json!(json_value))
 	});
 
-	io_handler.add_method("author_submitVCRequest", move |params: Params| {
-		debug!("worker_api_direct rpc was called: author_submitVCRequest");
+	io_handler.add_sync_method("author_requestVc", move |params: Params| {
+		debug!("worker_api_direct rpc was called: author_requestVc");
 
-		async move {
-			let json_value = match submit_vc_request_inner(params).await {
-				Ok(value) => value.to_hex(),
-				Err(error) => compute_hex_encoded_return_error(&error),
-			};
+		// TODO: optimise the error handling
+		let payload = match get_request_payload(params) {
+			Ok(v) => v,
+			Err(e) =>
+				return Ok(json!(compute_hex_encoded_return_error(
+					format!("Failed to get request payload: {}", e).as_str()
+				))),
+		};
+		let request = match AesRequest::from_hex(&payload) {
+			Ok(v) => v,
+			Err(e) =>
+				return Ok(json!(compute_hex_encoded_return_error(
+					format!("Failed to construct AesRequest: {:?}", e).as_str()
+				))),
+		};
 
-			Ok(json!(json_value))
+		let vc_request_sender = VcRequestSender::new();
+		let (sender, receiver) = channel::<Result<Vec<u8>, String>>();
```

Comment:
I see that we have replaced `oneshot` channel with `mpsc` channel, Does it make sense to use an MPSC channel here?

Reply:
> I see that we have replaced `oneshot` channel with `mpsc` channel, Does it make sense to use an MPSC channel here?

It is an mpsc channel :)

Reply:
So what I'm trying to say is we are using `mpsc` channel here over a `oneshot` channel even though we do not have multiple producers or require more than one message, In the above RPC function we have a single producer and a single consumer and a single message, all of which are attributes of a oneshot channel.

Reply:
Yes I believe either way would work, maybe even `Arc<Mutex<DequeVec<T>> ` works too.

I just wanted to keep it simple (and consistent) so that I didn't have to introduce extra crates 😂 - do you see any downsides using mpsc? Like more resource intensive?

Reply:
That's a good arg - I'll change it to oneshot channel